batch_seqr_loader/test/test_sm.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+
+"""
+Simulate communication with the SM server during the workflow
+"""
+
+import os
+import string
+import random
+from collections import defaultdict
+from typing import Union, List
+
+from sample_metadata import AnalysisUpdateModel
+from sample_metadata.api import SampleApi, AnalysisApi
+from sample_metadata.models.new_sample import NewSample
+from sample_metadata.models.analysis_model import AnalysisModel
+
+
+PROJ = os.environ.get('SM_DEV_DB_PROJECT', 'sm_dev')
+
+
+def _jc_pipeline_add_samples(test_run_id: str):
+    """
+    Add 3 samples: one with fastq input, one with CRAM input, one with GVCF input.
+    :param test_run_id: to suffix sample names for uniqueness
+    """
+    sapi = SampleApi()
+    s1 = NewSample(
+        external_id=f'NA12878-from-fq-{test_run_id}',
+        type='blood',
+        meta={
+            'reads': [
+                [
+                    'gs://cpg-seqr-test/batches/NA12878-trio-tiny/NA12878_L001_R1.fq',
+                    'gs://cpg-seqr-test/batches/NA12878-trio-tiny/NA12878_L002_R1.fq',
+                ],
+                [
+                    'gs://cpg-seqr-test/batches/NA12878-trio-tiny/NA12878_L001_R2.fq',
+                    'gs://cpg-seqr-test/batches/NA12878-trio-tiny/NA12878_L002_R2.fq',
+                ],
+            ]
+        },
+    )
+    s2 = NewSample(
+        external_id=f'NA12878-from-cram-{test_run_id}',
+        type='blood',
+        meta={'reads': 'gs://cpg-seqr-test/batches/NA12878-trio-tiny/NA12878.cram'},
+    )
+    s3 = NewSample(
+        external_id=f'NA12878-from-gvcf-{test_run_id}',
+        type='blood',
+        meta={'reads': 'gs://cpg-seqr-test/batches/NA12878-trio/NA12878.g.vcf.gz'},
+    )
+    sample_ids = [sapi.create_new_sample(PROJ, s) for s in (s1, s2, s3)]
+    print(f'Added samples {", ".join(sample_ids)}')
+    return sample_ids
+
+
+def _jc_pipeline_submit_analyses():
+    """
+    Add or update analyses. Iterate over completed analyses,
+    and submit next-step analyses
+    """
+
+    aapi = AnalysisApi()
+    print('aapi = AnalysisApi()')
+
+    # Get the list of latest complete analyses
+    latest_complete_analyses = aapi.get_latest_complete_analyses(project=PROJ)
+    print(f'Latest complete analyses: {latest_complete_analyses}')
+    latest_by_type_and_sids = defaultdict(list)
+    for a in latest_complete_analyses:
+        a_s_ids = sample_id_format(a['sample_ids'])
```

Comment:
The SM server should be doing this operation, I've sent a you a patch on slack :)

Reply:
Updated!